crypto/txscript/src/multi_sig.rs

```diff
@@ -0,0 +1,63 @@
+use crate::opcodes::to_small_int;
+use kaspa_consensus_core::{hashing::sighash::SigHashReusedValues, tx::VerifiableTransaction};
+use kaspa_txscript_errors::TxScriptError;
+
+use crate::opcodes::OpCodeImplementation;
+
+#[derive(Debug)]
+pub struct MultiSigScriptParameters {
+    pub required_signatures_count: u8,
+    pub signers_count: u8,
+    pub signers_pubkey: Vec<secp256k1::XOnlyPublicKey>,
+}
```

Comment:
that is only correct for schnorr based multisigs. so either add enum or suffix to make it cleat that it can only work with schnorr based mulisigs